app/javascript/mastodon/reducers/timelines.js

```diff
@@ -158,6 +160,8 @@ export default function timelines(state = initialState, action) {
     return filterTimelines(state, action.relationship, action.statuses);
   case ACCOUNT_UNFOLLOW_SUCCESS:
     return filterTimeline('home', state, action.relationship, action.statuses);
+  case HIDE_SUCCESS:
+    return deleteStatus(state, action.id, action.accountId, action.statuses);
```

Comment:
The last parameter of deleteStatus is a list of (statusId, accountId) of toots that are boosts of the target status. It seems like (see comment for the actions file) that you are passing all the known statuses here instead.